www/nginx/src/opnsense/mvc/app/controllers/OPNsense/Nginx/forms/location.xml

```diff
@@ -255,7 +255,7 @@
     <label>TLS SNI Forwarding</label>
     <type>checkbox</type>
     <advanced>true</advanced>
-    <help>Check this box, if you want the client SNI header to be used instead of your backend hostname. This settings overrides the configured hostname in the upstream configuration.</help>
+    <help>Check this box, if you want the client SNI header ($ssl_server_name) to be used as $proxy_ssl_name. Can be overriden in the upstream's hostname configuration.</help>
```

Comment:
This declines the quality of the help text actually. I don't have a suggestion for lack of knowledge but I don't want this in as is.

Reply:
Where else should I document these important technical details?

Reply:
"Enables passing of the server name through TLS Server Name Indication (SNI) when establishing a connection with Upstream (proxy_ssl_server_name)"?

Reply:
@fichtner Please provide me your preference.

Reply:
How about this:

... if you want to passthrough the hostname you get from the downstream to the upstream connection. ....

... Please note, that TLS server certificates are validated against this servername. If the server name passed by the client leads to different processing on the upstream server (i. e. different virtual host is choosen), this may cause security leaks.